content/blog/2018/06/2018-06-26-new-login-page.adoc

```diff
@@ -0,0 +1,173 @@
+      <st:include it="${h.simpleDecorator}" page="simple-head.jelly" optional="true"/>
+    </head>
+    <body>
+      <div class="simple-page" role="main">
+        <st:include it="${h.simpleDecorator}" page="simple-header.jelly" optional="true"/>
+      </div>
+      <div class="footer">
+        <st:include it="${h.simpleDecorator}" page="simple-footer.jelly" optional="true"/>
+      </div>
+    </body>
+  </html>
+</j:jelly>
+----
+
+The *sign-up* page only supports the simple-head.jelly:
+
+[source,html]
+----
+
+<?jelly escape-by-default='true'?>
+<j:jelly xmlns:j="jelly:core" xmlns:st="jelly:stapler" >
+  <j:new var="h" className="hudson.Functions"/>
+  <html>
+    <head>
+      <!-- css styling, will fallback to default implementation -->
+      <st:include it="${h.simpleDecorator}" page="simple-head.jelly" optional="true"/>
+    </head>
+  </html>
+</j:jelly>
+----
+
+== SimplePageDecorator - custom implementations
+
+NOTE: Have a look at https://github.com/daniel-beck/login-theme-plugin[Login Theme Plugin] (currently unreleased), which allows you to
+configure your own custom content to be injected into the new login/sign-up page.
+
+To allow easy customisation the decorator only implements one instance by the principal "first-come-first-serve".
+If jenkins finds an extension of the `SimplePageDecorator` it will use the Jelly files provided by that plugin.
+Otherwise Jenkins will fall back to the default implementation.
+
+[source,java]
+----
+@Extension
+public class MySimplePageDecorator extends SimplePageDecorator {
+   public String getProductName() {
+     return "MyJenkins";
+   }
+}
+----
+
+NOTE: The above will take override over the default because the default implementation has a very low ordinal (`@Extension(ordinal=-9999)`)
+If you have competing plugins implementing `SimplePageDecorator`, the implementation with the highest ordinal will be used.
+
+As a simple example, to customize the logo we display in the login page, create a `simple-head.jelly` with the following content:
+
+[source,html]
+----
+<?jelly escape-by-default='true'?>
+<j:jelly xmlns:j="jelly:core">
+  <link rel="stylesheet" href="${resURL}/css/simple-page.css" type="text/css" />
+  <link rel="stylesheet" href="${resURL}/css/simple-page.theme.css" type="text/css" />
+  <style>
+    .simple-page .logo {
+        background-image: url('${resURL}/plugin/YOUR_PLUGIN/icons/my.svg');
+        background-repeat: no-repeat;
+        background-position: 50% 0;
+        height: 130px;
+    }
+  </style>
+  <link rel="stylesheet" href="${resURL}/css/simple-page-forms.css" type="text/css" />
+</j:jelly>
+----
+
+To customize the login page further, create a `simple-header.jelly` like this:
+
+[source,html]
+----
+<?jelly escape-by-default='true'?>
+<j:jelly xmlns:j="jelly:core">
+  <div id="loginIntro">
+    <div class="logo"> </div>
+    <h1 id="productName">Welcome to ${it.productName}!</h1>
+  </div>
+</j:jelly>
+----
+
+The above technique is used to create:
```

Comment:
"For example, I used this technique to create a prototype of a login page for a CloudBees product I am working on"
 
Or so

Some summary after this image would also help